### ISSUE TITLE
cannot load the checker - does not recognize target language correctly

### ISSUE THREAD

**Issue by osher** [atom/spell-check#342]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

Does not recognize language correctly - takes naive assumptions about the relation between input language and locale.

### Steps to Reproduce

1. Use English as language of input
2. Use Israel as location (IL)
3. Use `Atom` with `spell-check`.
4. Start `Atom`

**Expected behavior:**

Should work.

**Actual behavior:**

I get on the top-right the following error message, and in settings page the plugin is paused:
```
The package spell-check cannot load the checker for en_IL. See the settings for ways of changing the languages used, resolving missing dictionaries, or hiding this warning.
```

IMHO - there is no such locale as `en_IL`

**Reproduces how often:**

Every time I start Atom

### Versions

```
$ atom --version
$ apm --version
apm  2.5.0
npm  6.14.5
node 10.20.1 x64
atom 1.48.0
python 2.7.17
git 2.17.1
```

Ubuntu 18.04

### Additional Information

Keep up the good work

**Comment by dmoonfire** [comment]
The code uses the `LANG` environment variable to [guess languages](https://github.com/atom/spell-check/blob/31296a49f3ba0d684284472e579425fdb00682f7/lib/spell-check-manager.coffee#L363-L367). Would you be so kind as to give the output from the following command:

```
echo $LANG
```

**Comment by dmoonfire** [comment]
Since there hasn't been any response for a while, I'm going to close this issue. Feel free to reopen.
